src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/MtAmazonDynamoDbStreamsBySharedTable.java

```diff
@@ -55,6 +54,31 @@ public GetShardIteratorResult getShardIterator(GetShardIteratorRequest request)
         return super.getShardIterator(request);
     }
 
+    @Override
+    protected MtGetRecordsResult getMtRecords(GetRecordsRequest request, StreamArn streamArn) {
```

Comment:
If we retrieve records without tenant context (KCL), then we don't need to filter anything, but we do need to use the prefix on each record to establish tenant context, so we can map each record using the virtual table definition.